alert-database/src/main/java/com/synopsys/integration/alert/database/api/StaticJobAccessor.java

```diff
@@ -183,40 +182,40 @@ public StaticJobAccessor(
         return getMatchingEnabledJobs(() -> distributionJobRepository.findMatchingEnabledJob(providerConfigId, notificationType.name()));
     }
 
+    // TODO this should be paged
     @Override
-    public List<FilteredDistributionJobModel> getMatchingEnabledJobs(FilteredDistributionJobRequestModel filteredDistributionJobRequestModel) {
-        FrequencyType frequencyType = filteredDistributionJobRequestModel.getFrequencyType();
-        NotificationType notificationType = filteredDistributionJobRequestModel.getNotificationType();
+    public List<FilteredDistributionJobResponseModel> getMatchingEnabledJobs(FilteredDistributionJobRequestModel filteredDistributionJobRequestModel) {
+        List<String> frequencyTypes = filteredDistributionJobRequestModel.getFrequencyTypes()
+                                          .stream()
+                                          .map(Enum::name)
+                                          .collect(Collectors.toList());
 
+        NotificationType notificationType = filteredDistributionJobRequestModel.getNotificationType();
         String projectName = filteredDistributionJobRequestModel.getProjectName();
 
         List<String> policyNames = filteredDistributionJobRequestModel.getPolicyNames();
         List<String> vulnerabilitySeverities = filteredDistributionJobRequestModel.getVulnerabilitySeverities();
 
         List<DistributionJobEntity> distributionJobEntities;
         if (!policyNames.isEmpty()) {
-            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobsWithPolicyNames(frequencyType.name(), notificationType.name(), projectName, policyNames);
+            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobsWithPolicyNames(frequencyTypes, notificationType.name(), projectName, policyNames);
         } else if (!vulnerabilitySeverities.isEmpty()) {
-            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobsWithVulnerabilitySeverities(frequencyType.name(), notificationType.name(), projectName, vulnerabilitySeverities);
+            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobsWithVulnerabilitySeverities(frequencyTypes, notificationType.name(), projectName, vulnerabilitySeverities);
         } else {
-            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobs(frequencyType.name(), notificationType.name(), projectName);
+            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobs(frequencyTypes, notificationType.name(), projectName);
         }
 
-        // TODO running project name pattern checks in java code, try to do this in SQL instead
+        // TODO running project name pattern checks in java code, try to do this in SQL instead (Won't need to return DistributionJobEntity anymore if this happens
```

Comment:
Doing this in SQL requires us to implement some JPA repositories, leaving it as an improvement in the future

Reply:
I don't think the project name pattern can matched in SQL. Java and SQL use different pattern formats that are not compatible.

Reply:
If we use a native query we could potentially pull it off: 
https://www.postgresql.org/docs/12/functions-matching.html#FUNCTIONS-SIMILARTO-REGEXP

Reply:
Ya the SIMILAR TO would be the functionality i'd be adding when implementing. If that doesn't work then I'll have to continue using the java filtering

Reply:
Gavin has pionted out that POSIX is needed over SIMILAR TO, but it still looks possible